src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar.kt

```diff
@@ -326,22 +268,21 @@ public abstract class ShadowJar :
     return clazz.isAnnotationPresent(CacheableTransformer::class.java)
   }
 
-  private fun configureRelocation() {
-    val packages = mutableSetOf<String>()
-    // Must cast configurations to List<FileCollection> to fix type mismatch in runtime.
-    (configurations as List<FileCollection>).forEach { configuration ->
-      configuration.files.forEach { jarFile ->
-        JarFile(jarFile).use { jf ->
-          jf.entries().asSequence().forEach { entry ->
-            if (entry.name.endsWith(".class") && entry.name != "module-info.class") {
-              packages.add(entry.name.substringBeforeLast('/').replace('/', '.'))
-            }
+  private val packageRelocators: List<SimpleRelocator>
+    get() {
+      if (!enableRelocation.get()) return emptyList()
+
+      val prefix = relocationPrefix.get()
+      // Must cast configurations to List<FileCollection> to fix type mismatch in runtime.
+      return (configurations.get() as List<FileCollection>).flatMap { configuration ->
+        configuration.files.flatMap { file ->
+          JarFile(file).use { jarFile ->
+            jarFile.entries().toList()
+              .filter { it.name.endsWith(".class") && it.name != "module-info.class" }
+              .map { it.name.substringBeforeLast('/').replace('/', '.') }
+              .map { SimpleRelocator(it, "$prefix.$it") }
```

Comment:
This is a regression, we have to reduce duplicates before creating relocators, fixed in #1271.